### PR TITLE
libkb: fixes for revoked subkeys

### PR DESCRIPTION
**Pull request by maxtaco**
- import properly and make sure we don't sign with revoked keys

### CODE REVIEW

**Comment by maxtaco** [comment]
r? @oconnor663 

Just a merge of keybase/go-crypto#15

File: go/libkb/keymerge.go [hunk 1]
```diff
@@ -57,6 +57,9 @@ func (to *PGPKeyBundle) MergeKey(from *PGPKeyBundle) {
 		if i, ok := existingSubkeys[subkey.PublicKey.Fingerprint]; ok {
 			if subkey.Sig.CreationTime.After(to.Subkeys[i].Sig.CreationTime) {
 				to.Subkeys[i].Sig = subkey.Sig
+				if subkey.Revocation != nil {
+					to.Subkeys[i].Revocation = subkey.Revocation
+				}
```
Comment (oconnor663) [inline go/libkb/keymerge.go]: Does anything need to change on line 49 above?

Comment (maxtaco) [inline go/libkb/keymerge.go]: Nope, I don't think so!

File: go/libkb/pgp_key.go [hunk 2]
```diff
@@ -255,6 +255,10 @@ func PGPKeyRawToArmored(raw []byte, priv bool) (ret string, err error) {
 	return
 }
 
+func (k *PGPKeyBundle) SerializePrivate(w io.Writer) error {
+	return k.Entity.SerializePrivate(w, &packet.Config{ReuseSignaturesOnSerialize: !k.Generated})
```
Comment (oconnor663) [inline go/libkb/pgp_key.go]: Out of curiosity, why does the generated case make more signatures during serialization?

File: go/libkb/skb.go [hunk 3]
```diff
@@ -24,6 +25,18 @@ import (
...
+	tmp, err := ioutil.TempFile(os.TempDir(), "dump-"+name)
+	if err != nil {
+		g.Log.Warning("Failed to dumpKey %s: %s", name, err)
+		return
+	}
+	g.Log.Notice("DUMPKEY %s -> %s", name, tmp.Name())
+	buf := bytes.NewBuffer(b)
+	io.Copy(tmp, buf)
+	tmp.Close()
+}
```
Comment (oconnor663) [inline go/libkb/skb.go]: Is this function just debugging code? Did you mean to check it in?

**Comment by oconnor663** [comment]
Looks good, just the questions above.

File: go/libkb/skb.go (hunk 3 above)
Comment (maxtaco) [inline go/libkb/skb.go]: but useful in the future....

File: go/libkb/pgp_key.go (hunk 2 above)
Comment (maxtaco) [inline go/libkb/pgp_key.go]: come again?

File: go/libkb/skb.go (hunk 3 above)
Comment (oconnor663) [inline go/libkb/skb.go]: If you want to keep it in, maybe call it something with Debug in the name, or leave a comment about it?

Comment (maxtaco) [inline go/libkb/skb.go]: ok!

File: go/libkb/pgp_key.go (hunk 2 above)
Comment (oconnor663) [inline go/libkb/pgp_key.go]: If `ReuseSignaturesOnSerialize` is false, remind me what happens? Do signatures just get dropped, or do new signatures also get made?

Comment (maxtaco) [inline go/libkb/pgp_key.go]: new sigs are made.  which need to happen in the generated case.

we generate an encryption subkey but not signing subkey, so we get away with it :/

**Pull request merged**